src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/data/GerritProject.java

```diff
@@ -216,32 +217,29 @@ public boolean isInteresting(String project, String branch, String topic, List<S
         if (compareType.matches(pattern, project)) {
             for (Branch b : branches) {
                 boolean foundInterestingForbidden = false;
-                boolean foundInterestingTopicOrFile = false;
                 if (b.isInteresting(branch)) {
                     if (forbiddenFilePaths != null) {
-                        for (FilePath ffp : forbiddenFilePaths) {
-                            if (ffp.isInteresting(files)) {
-                                foundInterestingForbidden = true;
-                                break;
+                        Iterator<String> i = files.iterator();
+                        while (i.hasNext()) {
+                            String file = i.next();
+                            for (FilePath ffp : forbiddenFilePaths) {
+                                if (ffp.isInteresting(file)) {
+                                    if (!disableStrictForbiddenFileVerification) {
+                                        return false;
+                                    } else {
+                                        foundInterestingForbidden = true;
+                                        i.remove();
```

Comment:
Potentially this line creates bug. Because https://github.com/sonyxperiadev/gerrit-events/blob/7f3547c6d55946e25e99a847b5160d69e59994ba/src/main/java/com/sonymobile/tools/gerrit/gerritevents/dto/events/ChangeBasedEvent.java caches the list of files and checks it against null. So, one trigger modifies list of files visible for another trigger.